GW2EIBuilders/RawFormatBuilder.cs

```diff
@@ -1,130 +1,71 @@
-            writer.Close();
-            JsonLog log = JsonConvert.DeserializeObject<JsonLog>(sw.ToString(), new JsonSerializerSettings
-            {
-                NullValueHandling = NullValueHandling.Ignore
-            });
-            return log;
+            throw new InvalidDataException("Missing settings in RawFormatBuilder");
         }
+        _jsonLog = JsonLogBuilder.BuildJsonLog(log, settings, parserVersion, uploadResults.ToArray());
+    }
 
-        /// <summary>
-        /// Creates a json file based on the original JsonLog of the RawFormat builder
-        /// </summary>
-        /// <param name="sw"></param>
-        /// <param name="indent"></param>
-        public void CreateJSON(StreamWriter sw, bool indent)
-        {
-            CreateJSON(_jsonLog, sw, indent);
-        }
+    /// <summary>
+    /// Returns a copy of JsonLog object that will be used by the builder.
+    /// </summary>
+    public JsonLog GetJson()
+    {
+        //TODO(Rennorb) @perf: What in the javascript is this...
+        return JsonSerializer.Deserialize<JsonLog>(JsonSerializer.Serialize(_jsonLog, SerializerSettings.Default)!, SerializerSettings.Default)!;
+    }
 
-        /// <summary>
-        /// Creates a json file based on the given JsonLog
-        /// </summary>
-        /// <param name="jsonLog"></param>
-        /// <param name="sw"></param>
-        /// <param name="indent"></param>
-        public static void CreateJSON(JsonLog jsonLog, StreamWriter sw, bool indent)
-        {
-            var serializer = new JsonSerializer
-            {
-                NullValueHandling = NullValueHandling.Ignore,
-                ContractResolver = DefaultJsonContractResolver,
-                StringEscapeHandling = StringEscapeHandling.EscapeHtml
-            };
-            var writer = new JsonTextWriter(sw)
-            {
-                Formatting = indent ? Newtonsoft.Json.Formatting.Indented : Newtonsoft.Json.Formatting.None
-            };
-            serializer.Serialize(writer, jsonLog);
-            writer.Close();
-        }
+    /// <summary>
+    /// Writes the original JsonLog of the RawFormat builder
+    /// </summary>
+    public void CreateJSON(Stream stream, bool indent)
+    {
+        CreateJSON(_jsonLog, stream, indent);
+    }
 
-        /// <summary>
-        /// Creates an xml file based on the original JsonLog of the RawFormat builder
-        /// </summary>
-        /// <param name="sw"></param>
-        /// <param name="indent"></param>
-        public void CreateXML(StreamWriter sw, bool indent)
-        {
-            CreateXML(_jsonLog, sw, indent);
-        }
+    /// <summary>
+    /// Writes the given JsonLog to the stream as json formatted text.
+    /// </summary>
+    public static void CreateJSON(JsonLog jsonLog, Stream stream, bool indent)
+    {
+        JsonSerializer.Serialize(stream, jsonLog, indent ? SerializerSettings.Indentent : SerializerSettings.Default);
```

Comment:
Does the serializer here also take into account the custom tuple converter?

Reply:
It does not, as the generator needs separate instructions per root type.
I did indeed forget to add converters for the JsonLog.

Reply:
Yeah, go for it.

Reply:
Added a new Tuple2ConverterFactory that replaces the old converter and added it to the JsonLog.

Quick question: why are there even two different json representations?

Reply:
One if officially documented and stable, the other one is unstable and is tailored in a way that matches the components of the html (it also does not contain data not exploited by the html, contrary to the documented JSON), so it changes anytime the components are moved around or reorganized.